core/plugins/org.fusesource.ide.camel.model.service.impl/src/org/fusesource/ide/camel/model/service/internal/DynamicCamelCatalog.java

```diff
@@ -0,0 +1,170 @@
+/******************************************************************************* 
+ * Copyright (c) 2018 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.fusesource.ide.camel.model.service.internal;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.camel.catalog.CamelCatalog;
+import org.apache.camel.catalog.DefaultCamelCatalog;
+import org.apache.camel.catalog.karaf.KarafRuntimeProvider;
+import org.apache.camel.catalog.maven.MavenVersionManager;
+import org.apache.camel.catalog.springboot.SpringBootRuntimeProvider;
+import org.fusesource.ide.camel.model.service.core.util.CamelCatalogUtils;
+import org.fusesource.ide.camel.model.service.impl.ICamelCatalogWrapper;
+import org.fusesource.ide.preferences.StagingRepositoriesUtils;
+import org.jboss.tools.foundation.core.plugin.log.IPluginLog;
+
+public class DynamicCamelCatalog implements ICamelCatalogWrapper {
+	
+	private static final String OLDER_CAMEL_VERSION_WITH_CATALOG = "2.15.0";
+	private static final String OLDER_CAMEL_CATALOG_VERSION_TO_LOAD_IN_CASE_OF_ERROR = "2.15.6";
+	
+	private CamelCatalog catalog;
+	private IPluginLog logger;
+	
+	public DynamicCamelCatalog(IPluginLog logger) {
+		this.logger = logger;
+		this.catalog = new DefaultCamelCatalog(true);
+		MavenVersionManager versionManager = new MavenVersionManager();
+		GrapeEnvironmentConfigurator grapeEnvironmentConfigurator = new GrapeEnvironmentConfigurator();
+		try {
+			versionManager.setCacheDirectory(grapeEnvironmentConfigurator.getGrapeFolderInsideTempFolder().toFile().getAbsolutePath());
+		} catch (IOException e) {
+			logger.logError(e);
+		}
+		configureAdditionalRepos(versionManager);
+		catalog.setVersionManager(versionManager);
+	}
+	
+	private void configureAdditionalRepos(MavenVersionManager versionManager) {
+		List<List<String>> additionalM2Repos = StagingRepositoriesUtils.getAdditionalRepos();
+		for (List<String> repo : additionalM2Repos) {
+			String repoName = repo.get(0);
+			String repoUri = repo.get(1);
+			versionManager.addMavenRepository(repoName, repoUri);
+		}
+	}
+
+	@Override
+	public String getLoadedVersion() {
+		return catalog.getLoadedVersion();
+	}
+
+	public String loadVersion(String requestedVersion) {
+		if (!catalog.loadVersion(requestedVersion)) {
+			if(OLDER_CAMEL_VERSION_WITH_CATALOG.compareTo(requestedVersion) > 1) {
```

Comment:
pretty sure we had a method somewhere that does this check already

Reply:
yes, it was in CamelService :-)
I moved this code from CamelService class: https://github.com/jbosstools/jbosstools-fuse/pull/1337/files#diff-6f3995e9cde8ad715b8ba14e9d12872aL113

or there was duplicated before that we can factorize?

Reply:
I think we have some duplicated code around camel version comparisons. How about moving Camel version specific util code to https://github.com/jbosstools/jbosstools-fuse/blob/master/core/plugins/org.fusesource.ide.foundation.core/src/org/fusesource/ide/foundation/core/util/VersionUtil.java instead?

Reply:
updated this old code to use the VersionUtil